### ISSUE TITLE
A required class was missing while executing on dependency-updates-report

### ISSUE THREAD

**Issue by sabaka** [mojohaus/versions-maven-plugin#177]
Hi,

After the latest version were released I'm unable to build dependency-updates-report.
Execution `mvn versions:dependency-updates-report` failed with following error:
```
[ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.4:dependency-updates-report (default-cli) on project checkstyle: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.4:dependency-updates-report failed: A required class was missing while executing org.codehaus.mojo:versions-maven-plugin:2.4:dependency-updates-report: org/apache/maven/doxia/sink/render/RenderingContext
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.codehaus.mojo:versions-maven-plugin:2.4
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/idubinin/.m2/repository/org/codehaus/mojo/versions-maven-plugin/2.4/versions-maven-plugin-2.4.jar
[ERROR] urls[1] = file:/home/idubinin/.m2/repository/backport-util-concurrent/backport-util-concurrent/3.1/backport-util-concurrent-3.1.jar
[ERROR] urls[2] = file:/home/idubinin/.m2/repository/org/slf4j/slf4j-jdk14/1.5.6/slf4j-jdk14-1.5.6.jar
[ERROR] urls[3] = file:/home/idubinin/.m2/repository/org/slf4j/jcl-over-slf4j/1.5.6/jcl-over-slf4j-1.5.6.jar
[ERROR] urls[4] = file:/home/idubinin/.m2/repository/commons-cli/commons-cli/1.2/commons-cli-1.2.jar
[ERROR] urls[5] = file:/home/idubinin/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[6] = file:/home/idubinin/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[7] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.11/plexus-interpolation-1.11.jar
[ERROR] urls[8] = file:/home/idubinin/.m2/repository/org/apache/maven/reporting/maven-reporting-api/3.0/maven-reporting-api-3.0.jar
[ERROR] urls[9] = file:/home/idubinin/.m2/repository/org/apache/maven/reporting/maven-reporting-impl/2.2/maven-reporting-impl-2.2.jar
[ERROR] urls[10] = file:/home/idubinin/.m2/repository/commons-validator/commons-validator/1.3.1/commons-validator-1.3.1.jar
[ERROR] urls[11] = file:/home/idubinin/.m2/repository/commons-beanutils/commons-beanutils/1.7.0/commons-beanutils-1.7.0.jar
[ERROR] urls[12] = file:/home/idubinin/.m2/repository/commons-digester/commons-digester/1.6/commons-digester-1.6.jar
[ERROR] urls[13] = file:/home/idubinin/.m2/repository/xml-apis/xml-apis/1.0.b2/xml-apis-1.0.b2.jar
[ERROR] urls[14] = file:/home/idubinin/.m2/repository/commons-logging/commons-logging/1.0.4/commons-logging-1.0.4.jar
[ERROR] urls[15] = file:/home/idubinin/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar
[ERROR] urls[16] = file:/home/idubinin/.m2/repository/org/sonatype/sisu/sisu-inject-bean/1.4.2/sisu-inject-bean-1.4.2.jar
[ERROR] urls[17] = file:/home/idubinin/.m2/repository/org/sonatype/sisu/sisu-guice/2.1.7/sisu-guice-2.1.7-noaop.jar
[ERROR] urls[18] = file:/home/idubinin/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.1.0/maven-shared-utils-3.1.0.jar
[ERROR] urls[19] = file:/home/idubinin/.m2/repository/commons-io/commons-io/2.5/commons-io-2.5.jar
[ERROR] urls[20] = file:/home/idubinin/.m2/repository/org/apache/maven/wagon/wagon-file/2.12/wagon-file-2.12.jar
[ERROR] urls[21] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-core/1.7/doxia-core-1.7.jar
[ERROR] urls[22] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-logging-api/1.7/doxia-logging-api-1.7.jar
[ERROR] urls[23] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[24] = file:/home/idubinin/.m2/repository/org/apache/httpcomponents/httpclient/4.0.2/httpclient-4.0.2.jar
[ERROR] urls[25] = file:/home/idubinin/.m2/repository/commons-codec/commons-codec/1.3/commons-codec-1.3.jar
[ERROR] urls[26] = file:/home/idubinin/.m2/repository/org/apache/httpcomponents/httpcore/4.0.1/httpcore-4.0.1.jar
[ERROR] urls[27] = file:/home/idubinin/.m2/repository/xmlunit/xmlunit/1.5/xmlunit-1.5.jar
[ERROR] urls[28] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-sink-api/1.7/doxia-sink-api-1.7.jar
[ERROR] urls[29] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-site-renderer/1.7/doxia-site-renderer-1.7.jar
[ERROR] urls[30] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-decoration-model/1.7/doxia-decoration-model-1.7.jar
[ERROR] urls[31] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-skin-model/1.7/doxia-skin-model-1.7.jar
[ERROR] urls[32] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-module-xhtml/1.7/doxia-module-xhtml-1.7.jar
[ERROR] urls[33] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-module-fml/1.7/doxia-module-fml-1.7.jar
[ERROR] urls[34] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-velocity/1.1.7/plexus-velocity-1.1.7.jar
[ERROR] urls[35] = file:/home/idubinin/.m2/repository/org/apache/velocity/velocity/1.7/velocity-1.7.jar
[ERROR] urls[36] = file:/home/idubinin/.m2/repository/org/apache/velocity/velocity-tools/2.0/velocity-tools-2.0.jar
[ERROR] urls[37] = file:/home/idubinin/.m2/repository/commons-chain/commons-chain/1.1/commons-chain-1.1.jar
[ERROR] urls[38] = file:/home/idubinin/.m2/repository/dom4j/dom4j/1.1/dom4j-1.1.jar
[ERROR] urls[39] = file:/home/idubinin/.m2/repository/oro/oro/2.0.8/oro-2.0.8.jar
[ERROR] urls[40] = file:/home/idubinin/.m2/repository/sslext/sslext/1.2-0/sslext-1.2-0.jar
[ERROR] urls[41] = file:/home/idubinin/.m2/repository/org/apache/struts/struts-core/1.3.8/struts-core-1.3.8.jar
[ERROR] urls[42] = file:/home/idubinin/.m2/repository/antlr/antlr/2.7.2/antlr-2.7.2.jar
[ERROR] urls[43] = file:/home/idubinin/.m2/repository/org/apache/struts/struts-taglib/1.3.8/struts-taglib-1.3.8.jar
[ERROR] urls[44] = file:/home/idubinin/.m2/repository/org/apache/struts/struts-tiles/1.3.8/struts-tiles-1.3.8.jar
[ERROR] urls[45] = file:/home/idubinin/.m2/repository/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar
[ERROR] urls[46] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar
[ERROR] urls[47] = file:/home/idubinin/.m2/repository/org/apache/xbean/xbean-reflect/3.7/xbean-reflect-3.7.jar
[ERROR] urls[48] = file:/home/idubinin/.m2/repository/com/google/collections/google-collections/1.0/google-collections-1.0.jar
[ERROR] urls[49] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-interactivity-api/1.0-alpha-6/plexus-interactivity-api-1.0-alpha-6.jar
[ERROR] urls[50] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-i18n/1.0-beta-10/plexus-i18n-1.0-beta-10.jar
[ERROR] urls[51] = file:/home/idubinin/.m2/repository/org/codehaus/woodstox/woodstox-core-asl/4.2.0/woodstox-core-asl-4.2.0.jar
[ERROR] urls[52] = file:/home/idubinin/.m2/repository/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2.jar
[ERROR] urls[53] = file:/home/idubinin/.m2/repository/org/codehaus/woodstox/stax2-api/3.1.1/stax2-api-3.1.1.jar
[ERROR] urls[54] = file:/home/idubinin/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[project>com.puppycrawl.tools:checkstyle:8.1-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : org.apache.maven.doxia.sink.render.RenderingContext
[ERROR] -> [Help 1]
```

**Comment by khmarbaise** [comment]
Can you please first delete the folder `$HOME/.m2/repository/org/apache/doxia` and retry that ?

**Comment by sabaka** [comment]
I've removed `$HOME/.m2/repository/org/apache/maven/doxia`. Unfortunately it didn't help :(

```
[master ↑·113|✔] $ tree $HOME/.m2/repository | grep doxia
[master ↑·113|✔] $ mvn versions:dependency-updates-report
......
[ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.4:dependency-updates-report (default-cli) on project checkstyle: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.4:dependency-updates-report failed: A required class was missing while executing org.codehaus.mojo:versions-maven-plugin:2.4:dependency-updates-report: org/apache/maven/doxia/sink/render/RenderingContext
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.codehaus.mojo:versions-maven-plugin:2.4
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/idubinin/.m2/repository/org/codehaus/mojo/versions-maven-plugin/2.4/versions-maven-plugin-2.4.jar
[ERROR] urls[1] = file:/home/idubinin/.m2/repository/backport-util-concurrent/backport-util-concurrent/3.1/backport-util-concurrent-3.1.jar
[ERROR] urls[2] = file:/home/idubinin/.m2/repository/org/slf4j/slf4j-jdk14/1.5.6/slf4j-jdk14-1.5.6.jar
[ERROR] urls[3] = file:/home/idubinin/.m2/repository/org/slf4j/jcl-over-slf4j/1.5.6/jcl-over-slf4j-1.5.6.jar
[ERROR] urls[4] = file:/home/idubinin/.m2/repository/commons-cli/commons-cli/1.2/commons-cli-1.2.jar
[ERROR] urls[5] = file:/home/idubinin/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.3/plexus-sec-dispatcher-1.3.jar
[ERROR] urls[6] = file:/home/idubinin/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[7] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.11/plexus-interpolation-1.11.jar
[ERROR] urls[8] = file:/home/idubinin/.m2/repository/org/apache/maven/reporting/maven-reporting-api/3.0/maven-reporting-api-3.0.jar
[ERROR] urls[9] = file:/home/idubinin/.m2/repository/org/apache/maven/reporting/maven-reporting-impl/2.2/maven-reporting-impl-2.2.jar
[ERROR] urls[10] = file:/home/idubinin/.m2/repository/commons-validator/commons-validator/1.3.1/commons-validator-1.3.1.jar
[ERROR] urls[11] = file:/home/idubinin/.m2/repository/commons-beanutils/commons-beanutils/1.7.0/commons-beanutils-1.7.0.jar
[ERROR] urls[12] = file:/home/idubinin/.m2/repository/commons-digester/commons-digester/1.6/commons-digester-1.6.jar
[ERROR] urls[13] = file:/home/idubinin/.m2/repository/xml-apis/xml-apis/1.0.b2/xml-apis-1.0.b2.jar
[ERROR] urls[14] = file:/home/idubinin/.m2/repository/commons-logging/commons-logging/1.0.4/commons-logging-1.0.4.jar
[ERROR] urls[15] = file:/home/idubinin/.m2/repository/org/apache/maven/shared/maven-common-artifact-filters/3.0.1/maven-common-artifact-filters-3.0.1.jar
[ERROR] urls[16] = file:/home/idubinin/.m2/repository/org/sonatype/sisu/sisu-inject-bean/1.4.2/sisu-inject-bean-1.4.2.jar
[ERROR] urls[17] = file:/home/idubinin/.m2/repository/org/sonatype/sisu/sisu-guice/2.1.7/sisu-guice-2.1.7-noaop.jar
[ERROR] urls[18] = file:/home/idubinin/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.1.0/maven-shared-utils-3.1.0.jar
[ERROR] urls[19] = file:/home/idubinin/.m2/repository/commons-io/commons-io/2.5/commons-io-2.5.jar
[ERROR] urls[20] = file:/home/idubinin/.m2/repository/org/apache/maven/wagon/wagon-file/2.12/wagon-file-2.12.jar
[ERROR] urls[21] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-core/1.7/doxia-core-1.7.jar
[ERROR] urls[22] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-logging-api/1.7/doxia-logging-api-1.7.jar
[ERROR] urls[23] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-component-annotations/1.5.5/plexus-component-annotations-1.5.5.jar
[ERROR] urls[24] = file:/home/idubinin/.m2/repository/org/apache/httpcomponents/httpclient/4.0.2/httpclient-4.0.2.jar
[ERROR] urls[25] = file:/home/idubinin/.m2/repository/commons-codec/commons-codec/1.3/commons-codec-1.3.jar
[ERROR] urls[26] = file:/home/idubinin/.m2/repository/org/apache/httpcomponents/httpcore/4.0.1/httpcore-4.0.1.jar
[ERROR] urls[27] = file:/home/idubinin/.m2/repository/xmlunit/xmlunit/1.5/xmlunit-1.5.jar
[ERROR] urls[28] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-sink-api/1.7/doxia-sink-api-1.7.jar
[ERROR] urls[29] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-site-renderer/1.7/doxia-site-renderer-1.7.jar
[ERROR] urls[30] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-decoration-model/1.7/doxia-decoration-model-1.7.jar
[ERROR] urls[31] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-skin-model/1.7/doxia-skin-model-1.7.jar
[ERROR] urls[32] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-module-xhtml/1.7/doxia-module-xhtml-1.7.jar
[ERROR] urls[33] = file:/home/idubinin/.m2/repository/org/apache/maven/doxia/doxia-module-fml/1.7/doxia-module-fml-1.7.jar
[ERROR] urls[34] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-velocity/1.1.7/plexus-velocity-1.1.7.jar
[ERROR] urls[35] = file:/home/idubinin/.m2/repository/org/apache/velocity/velocity/1.7/velocity-1.7.jar
[ERROR] urls[36] = file:/home/idubinin/.m2/repository/org/apache/velocity/velocity-tools/2.0/velocity-tools-2.0.jar
[ERROR] urls[37] = file:/home/idubinin/.m2/repository/commons-chain/commons-chain/1.1/commons-chain-1.1.jar
[ERROR] urls[38] = file:/home/idubinin/.m2/repository/dom4j/dom4j/1.1/dom4j-1.1.jar
[ERROR] urls[39] = file:/home/idubinin/.m2/repository/oro/oro/2.0.8/oro-2.0.8.jar
[ERROR] urls[40] = file:/home/idubinin/.m2/repository/sslext/sslext/1.2-0/sslext-1.2-0.jar
[ERROR] urls[41] = file:/home/idubinin/.m2/repository/org/apache/struts/struts-core/1.3.8/struts-core-1.3.8.jar
[ERROR] urls[42] = file:/home/idubinin/.m2/repository/antlr/antlr/2.7.2/antlr-2.7.2.jar
[ERROR] urls[43] = file:/home/idubinin/.m2/repository/org/apache/struts/struts-taglib/1.3.8/struts-taglib-1.3.8.jar
[ERROR] urls[44] = file:/home/idubinin/.m2/repository/org/apache/struts/struts-tiles/1.3.8/struts-tiles-1.3.8.jar
[ERROR] urls[45] = file:/home/idubinin/.m2/repository/commons-collections/commons-collections/3.2.1/commons-collections-3.2.1.jar
[ERROR] urls[46] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar
[ERROR] urls[47] = file:/home/idubinin/.m2/repository/org/apache/xbean/xbean-reflect/3.7/xbean-reflect-3.7.jar
[ERROR] urls[48] = file:/home/idubinin/.m2/repository/com/google/collections/google-collections/1.0/google-collections-1.0.jar
[ERROR] urls[49] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-interactivity-api/1.0-alpha-6/plexus-interactivity-api-1.0-alpha-6.jar
[ERROR] urls[50] = file:/home/idubinin/.m2/repository/org/codehaus/plexus/plexus-i18n/1.0-beta-10/plexus-i18n-1.0-beta-10.jar
[ERROR] urls[51] = file:/home/idubinin/.m2/repository/org/codehaus/woodstox/woodstox-core-asl/4.2.0/woodstox-core-asl-4.2.0.jar
[ERROR] urls[52] = file:/home/idubinin/.m2/repository/javax/xml/stream/stax-api/1.0-2/stax-api-1.0-2.jar
[ERROR] urls[53] = file:/home/idubinin/.m2/repository/org/codehaus/woodstox/stax2-api/3.1.1/stax2-api-3.1.1.jar
[ERROR] urls[54] = file:/home/idubinin/.m2/repository/commons-lang/commons-lang/2.6/commons-lang-2.6.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[project>com.puppycrawl.tools:checkstyle:8.1-SNAPSHOT, parent: ClassRealm[maven.api, parent: null]]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : org.apache.maven.doxia.sink.render.RenderingContext
```

**Comment by sabaka** [comment]
@khmarbaise , do you have any clue why it could happen?
Maybe I can help somehow?

**Comment by Zero3** [comment]
I'm seeing the exact same problem with a fresh test project :(

**Comment by khmarbaise** [comment]
@sabaka Any help is appreciated..

**Comment by zman0900** [comment]
Seems very similar to this: https://github.com/mojohaus/license-maven-plugin/issues/73

Adding plugin dependency on different versions of doxia-core does change what error is produced, but does not fix the issue in this case.

**Comment by sabaka** [comment]
@khmarbaise , do you have any estimated release date?

**Comment by ecararus** [comment]
Can be released version 2.5 ?
